### PR TITLE
fix(action-popover): submenu is left aligned in the 'with submenu aligned right' story

### PR DESCRIPTION
**Pull request by mihai-albu-sage**
### Proposed behaviour

Reduce the left margin in the Action Popover story [Kayboard Navigation Right Aligned Submenu](https://carbon.sage.com/?path=/story/action-popover--keyboard-navigation-right-aligned-submenu) so that the submenu correctly aligns to the right.

### Current behaviour

The submenu aligns to the left.

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- go to the storybook page.
- open the Action Popover storybook named: Keyboard Navigation Right Aligned Submenu.
- notice the submenu is left aligned.

### CODE REVIEW

File: src/components/action-popover/action-popover.stories.tsx [hunk 1]
```diff
@@ -436,7 +436,7 @@ KeyboardNavigationLeftAlignedSubmenu.storyName =
 export const KeyboardNavigationRightAlignedSubmenu: Story = () => {
   return (
     <Box height={250}>
-      <ActionPopover>
+      <ActionPopover ml={0}>
```
Comment (DipperTheDan) [inline src/components/action-popover/action-popover.stories.tsx:439]: **nitpick(non-blocking)**: I think you might also need to add `submenuPosition="right"` prop here too.

Comment (DipperTheDan) [reply src/components/action-popover/action-popover.stories.tsx:439]: Forgot to add the screenshot. This is how it currently looks in the `docs` page on your PR. 
Screenshot 2024-04-16 at 16 18 14

**Comment by Parsium** [comment]
Nothing more to add to @DipperTheDan's comments. Happy to approve once we've addressed them 👍🏼

File: src/components/action-popover/action-popover.stories.tsx (hunk 1 above)
Comment (mihai-albu-sage) [reply src/components/action-popover/action-popover.stories.tsx]: Hi,

The `docs` page has been fixed.

Screenshot 2024-05-09 084858

**Review by DipperTheDan — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by divyajindel — COMMENTED** [review]
Hi @mihai-albu-sage , There is an arrow alignment issue with safari, please have a look once.

**Review by divyajindel — CHANGES_REQUESTED** [review]
Screenshot 2024-05-14 at 15 14 59

**Comment by DipperTheDan** [comment]
> Screenshot 2024-05-14 at 15 14 59
> Hi @mihai-albu-sage , There is an arrow alignment issue with safari, please have a look once.

@divyajindel This is already present on the master version of ActionPopover, specifically the one [With submenu positioned to right](https://carbon.sage.com/?path=/docs/action-popover--docs#with-submenu-positioned-to-right) example. Might be worth addressing this in a separate ticket seeing as @mihai-albu-sage hasn't introduced this bug.

Screenshot 2024-05-14 at 16 37 55

**Review by divyajindel — APPROVED** [review]

**Pull request merged**